bench/rttanalysis: TestBenchmarkExpectation failed

bench/rttanalysis.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15105198?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15105198?buildTab=artifacts#/) on master @ [b426d118ab1f78b9377c7e5f04bcfe166006eba6](https://github.com/cockroachdb/cockroach/commits/b426d118ab1f78b9377c7e5f04bcfe166006eba6):


```
    validate_benchmark_data.go:123: success: expected Jobs/show_job to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:123: success: expected ORMQueries/django_table_introspection_8_tables to perform KV lookups in [6, 6], got 5
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_class to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_type to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:120: fail: expected Jobs/jobs_page_type_filtered to perform KV lookups in [8, 8], got 6
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_attribute to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_my_temp_schema_multiple_times to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:123: success: expected ShowGrants/grant_4_roles to perform KV lookups in [18, 18], got 18
    validate_benchmark_data.go:123: success: expected ORMQueries/has_sequence_privilege to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ORMQueries/django_table_introspection_1_table to perform KV lookups in [6, 6], got 5
    validate_benchmark_data.go:123: success: expected Jobs/jobs_page_default to perform KV lookups in [8, 8], got 9
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_is_other_temp_schema to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:123: success: expected ShowGrants/grant_2_roles to perform KV lookups in [14, 14], got 14
    validate_benchmark_data.go:123: success: expected ORMQueries/django_column_introspection_4_tables to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected ORMQueries/hasura_column_descriptions to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:123: success: expected ORMQueries/django_column_introspection_8_tables to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected ORMQueries/has_schema_privilege to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ShowGrants/grant_3_roles to perform KV lookups in [16, 16], got 16
    validate_benchmark_data.go:123: success: expected ORMQueries/hasura_column_descriptions_modified to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:123: success: expected ORMQueries/hasura_column_descriptions_8_tables to perform KV lookups in [12, 12], got 12
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_is_other_temp_schema_multiple_times to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:123: success: expected ORMQueries/has_column_privilege_using_attnum to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ORMQueries/has_table_privilege to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ORMQueries/has_column_privilege_using_column_name to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected Jobs/jobs_page_type_filtered_no_matches to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:123: success: expected ORMQueries/introspection_description_join to perform KV lookups in [4, 4], got 5
    validate_benchmark_data.go:123: success: expected ORMQueries/information_schema._pg_index_position to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_my_temp_schema to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:123: success: expected AlterTableDropConstraint/alter_table_drop_1_check_constraint to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:123: success: expected Jobs/show_jobs to perform KV lookups in [6, 8], got 8
    validate_benchmark_data.go:123: success: expected ORMQueries/npgsql_fields to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:123: success: expected ORMQueries/npgsql_types to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:123: success: expected ORMQueries/prisma_column_descriptions_updated to perform KV lookups in [3, 3], got 4
    validate_benchmark_data.go:123: success: expected ORMQueries/liquibase_migrations to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected ORMQueries/column_descriptions_json_agg to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ORMQueries/liquibase_migrations_on_multiple_dbs to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:123: success: expected ORMQueries/prisma_column_descriptions to perform KV lookups in [133, 133], got 133
    validate_benchmark_data.go:123: success: expected ORMQueries/asyncpg_types to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:76: see the -rewrite flag to re-run the benchmarks and adjust the expectations
    validate_benchmark_data.go:77: usage: ./dev test --rewrite pkg/bench/rttanalysis -f=TestBenchmarkExpectation/ --test-args '-rewrite-iterations=N'
    validate_benchmark_data.go:110: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b03b0378e25aebc8fcd368039859e7a0/logTestBenchmarkExpectation1274490437
--- FAIL: TestBenchmarkExpectation (257.46s)
--- FAIL: TestBenchmarkExpectation (258.21s)
=== RUN   TestBenchmarkExpectation/Jobs/jobs_page_type_filtered
    rtt_analysis_bench.go:207: Jobs/jobs_page_type_filtered: got 6, expected 8
    rtt_analysis_bench.go:213: wrote jaeger trace to /artifacts/tmp/_tmp/b03b0378e25aebc8fcd368039859e7a0/logTestBenchmarkExpectation1274490437/Jobs_jobs_page_type_filtered.jaeger.json
        --- FAIL: TestBenchmarkExpectation/Jobs/jobs_page_type_filtered (143.83s)
=== RUN   TestBenchmarkExpectation/Jobs
    --- FAIL: TestBenchmarkExpectation/Jobs (166.52s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38413

bench/rttanalysis.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15139759?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15139759?buildTab=artifacts#/) on master @ [367b681abf1766e96f5768f7ba3ac7e9e696c99d](https://github.com/cockroachdb/cockroach/commits/367b681abf1766e96f5768f7ba3ac7e9e696c99d):


```
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_namespace to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:123: success: expected ORMQueries/django_column_introspection_1_table to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected Revoke/revoke_all_on_2_tables to perform KV lookups in [19, 19], got 19
    validate_benchmark_data.go:123: success: expected ORMQueries/liquibase_migrations to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_my_temp_schema to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:123: success: expected ORMQueries/django_column_introspection_8_tables to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected ORMQueries/django_column_introspection_4_tables to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected ORMQueries/django_table_introspection_1_table to perform KV lookups in [6, 6], got 5
    validate_benchmark_data.go:123: success: expected GenerateObjects/generate_100_tables_from_template to perform KV lookups in [11, 11], got 12
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_my_temp_schema_multiple_times to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:123: success: expected ORMQueries/has_sequence_privilege to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected Jobs/cancel_job to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:123: success: expected AlterTableAddForeignKey/alter_table_add_1_foreign_key to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:123: success: expected AlterTableAddForeignKey/alter_table_add_foreign_key_with_3_columns to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:123: success: expected AlterRole/alter_role_with_3_options to perform KV lookups in [29, 29], got 29
    validate_benchmark_data.go:123: success: expected GenerateObjects/generate_10_tables to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:123: success: expected AlterTableAddForeignKey/alter_table_add_2_foreign_keys to perform KV lookups in [19, 19], got 19
    validate_benchmark_data.go:123: success: expected GenerateObjects/generate_50000_tables to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:123: success: expected GenerateObjects/generate_1000_tables_-_this_test_should_use_the_same_number_of_RTTs_as_for_10_tables to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:123: success: expected AlterTableAddCheckConstraint/alter_table_add_1_check_constraint to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:123: success: expected AlterTableAddCheckConstraint/alter_table_add_2_check_constraints to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:123: success: expected ORMQueries/has_table_privilege to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected DropTable/drop_2_tables to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:123: success: expected Discard/DISCARD_ALL,_no_tables to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected AlterRole/alter_role_with_1_option to perform KV lookups in [18, 18], got 18
    validate_benchmark_data.go:123: success: expected Discard/DISCARD_ALL,_1_tables_in_1_db to perform KV lookups in [16, 16], got 16
    validate_benchmark_data.go:123: success: expected AlterTableAddForeignKey/alter_table_add_3_foreign_keys to perform KV lookups in [21, 21], got 21
    validate_benchmark_data.go:123: success: expected Discard/DISCARD_ALL,_2_tables_in_2_dbs to perform KV lookups in [23, 23], got 23
    validate_benchmark_data.go:123: success: expected GrantRole/grant_2_roles to perform KV lookups in [25, 25], got 25
    validate_benchmark_data.go:123: success: expected ORMQueries/django_table_introspection_8_tables to perform KV lookups in [6, 6], got 5
    validate_benchmark_data.go:123: success: expected GrantRole/grant_1_role to perform KV lookups in [19, 19], got 19
    validate_benchmark_data.go:123: success: expected Jobs/resume_job to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:123: success: expected Jobs/pause_job to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:123: success: expected ORMQueries/column_descriptions_json_agg to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected GenerateObjects/generate_10x10_schemas_and_tables_in_existing_db to perform KV lookups in [16, 16], got 16
    validate_benchmark_data.go:123: success: expected ORMQueries/liquibase_migrations_on_multiple_dbs to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:123: success: expected ORMQueries/prisma_column_descriptions to perform KV lookups in [133, 133], got 133
    validate_benchmark_data.go:123: success: expected ORMQueries/asyncpg_types to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:76: see the -rewrite flag to re-run the benchmarks and adjust the expectations
    validate_benchmark_data.go:77: usage: ./dev test --rewrite pkg/bench/rttanalysis -f=TestBenchmarkExpectation/ --test-args '-rewrite-iterations=N'
    validate_benchmark_data.go:110: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b03b0378e25aebc8fcd368039859e7a0/logTestBenchmarkExpectation2829641793
--- FAIL: TestBenchmarkExpectation (218.79s)
=== RUN   TestBenchmarkExpectation/Jobs
    --- FAIL: TestBenchmarkExpectation/Jobs (166.38s)
=== RUN   TestBenchmarkExpectation/Jobs/show_jobs
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    rtt_analysis_bench.go:207: Jobs/show_jobs: got 9, expected 6-8
    rtt_analysis_bench.go:213: wrote jaeger trace to /artifacts/tmp/_tmp/b03b0378e25aebc8fcd368039859e7a0/logTestBenchmarkExpectation2829641793/Jobs_show_jobs.jaeger.json
        --- FAIL: TestBenchmarkExpectation/Jobs/show_jobs (89.21s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123691 bench/rttanalysis: TestBenchmarkExpectation failed [C-test-failure O-robot T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

bench/rttanalysis.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15153717?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15153717?buildTab=artifacts#/) on master @ [8354defae95fe0f0ce8a98fc354980aa6a4294f5](https://github.com/cockroachdb/cockroach/commits/8354defae95fe0f0ce8a98fc354980aa6a4294f5):


```
    validate_benchmark_data.go:123: success: expected ORMQueries/introspection_description_join to perform KV lookups in [4, 4], got 5
    validate_benchmark_data.go:123: success: expected ORMQueries/has_table_privilege to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ORMQueries/has_sequence_privilege to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ORMQueries/has_column_privilege_using_attnum to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_my_temp_schema_multiple_times to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_is_other_temp_schema_multiple_times to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:123: success: expected ORMQueries/has_schema_privilege to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ORMQueries/has_column_privilege_using_column_name to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ORMQueries/hasura_column_descriptions_modified to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:123: success: expected ORMQueries/pg_is_other_temp_schema to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:123: success: expected ORMQueries/hasura_column_descriptions to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:120: fail: expected Jobs/jobs_page_type_filtered_no_matches to perform KV lookups in [8, 8], got 5
    validate_benchmark_data.go:123: success: expected ORMQueries/hasura_column_descriptions_8_tables to perform KV lookups in [12, 12], got 12
    validate_benchmark_data.go:123: success: expected ORMQueries/information_schema._pg_index_position to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected Discard/DISCARD_ALL,_no_tables to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected GenerateObjects/generate_10x10_schemas_and_tables_in_existing_db to perform KV lookups in [16, 16], got 16
    validate_benchmark_data.go:123: success: expected ORMQueries/npgsql_types to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:123: success: expected ORMQueries/npgsql_fields to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:123: success: expected Discard/DISCARD_ALL,_1_tables_in_1_db to perform KV lookups in [16, 16], got 16
    validate_benchmark_data.go:123: success: expected Discard/DISCARD_ALL,_2_tables_in_2_dbs to perform KV lookups in [23, 23], got 23
    validate_benchmark_data.go:123: success: expected AlterTableConfigureZone/alter_table_configure_zone_ranges to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:123: success: expected AlterTableConfigureZone/alter_table_configure_zone_5_replicas to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:123: success: expected ORMQueries/prisma_column_descriptions_updated to perform KV lookups in [3, 3], got 4
    validate_benchmark_data.go:123: success: expected AlterTableConfigureZone/alter_table_configure_zone_7_replicas_ to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:123: success: expected ORMQueries/liquibase_migrations to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:123: success: expected SystemDatabaseQueries/select_system.users_with_schema_Name to perform KV lookups in [1, 1], got 1
    validate_benchmark_data.go:123: success: expected AlterTableAddCheckConstraint/alter_table_add_1_check_constraint to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:123: success: expected SystemDatabaseQueries/select_system.users_without_schema_Name to perform KV lookups in [1, 1], got 1
    validate_benchmark_data.go:123: success: expected AlterTableUnsplit/alter_table_unsplit_at_3_values to perform KV lookups in [11, 11], got 11
    validate_benchmark_data.go:123: success: expected DropView/drop_2_views to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:123: success: expected DropSequence/drop_1_sequence to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:123: success: expected DropSequence/drop_2_sequences to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:123: success: expected CreateRole/create_role_with_1_option to perform KV lookups in [26, 26], got 26
    validate_benchmark_data.go:123: success: expected AlterTableAddForeignKey/alter_table_add_2_foreign_keys to perform KV lookups in [19, 19], got 19
    validate_benchmark_data.go:123: success: expected Jobs/resume_job to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:123: success: expected ORMQueries/column_descriptions_json_agg to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:123: success: expected ORMQueries/liquibase_migrations_on_multiple_dbs to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:123: success: expected ORMQueries/prisma_column_descriptions to perform KV lookups in [133, 133], got 133
    validate_benchmark_data.go:123: success: expected ORMQueries/asyncpg_types to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:76: see the -rewrite flag to re-run the benchmarks and adjust the expectations
    validate_benchmark_data.go:77: usage: ./dev test --rewrite pkg/bench/rttanalysis -f=TestBenchmarkExpectation/ --test-args '-rewrite-iterations=N'
    validate_benchmark_data.go:110: -- test log scope end --
--- FAIL: TestBenchmarkExpectation (269.36s)
=== RUN   TestBenchmarkExpectation/Jobs/jobs_page_type_filtered_no_matches
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    rtt_analysis_bench.go:207: Jobs/jobs_page_type_filtered_no_matches: got 5.5, expected 8
    rtt_analysis_bench.go:213: wrote jaeger trace to /artifacts/tmp/_tmp/b03b0378e25aebc8fcd368039859e7a0/logTestBenchmarkExpectation3414327826/Jobs_jobs_page_type_filtered_no_matches.jaeger.json
        --- FAIL: TestBenchmarkExpectation/Jobs/jobs_page_type_filtered_no_matches (173.88s)
=== RUN   TestBenchmarkExpectation/Jobs
    --- FAIL: TestBenchmarkExpectation/Jobs (205.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #123691 bench/rttanalysis: TestBenchmarkExpectation failed [C-test-failure O-robot P-1 T-sql-foundations branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/bench/rttanalysis/testdata/benchmark_expectations
+++ b/pkg/bench/rttanalysis/testdata/benchmark_expectations
@@ -67,10 +67,10 @@ exp,benchmark
 6,Jobs/resume_job
 8,Jobs/jobs_page_default
 8,Jobs/jobs_page_latest_50
-8,Jobs/jobs_page_type_filtered
-8,Jobs/jobs_page_type_filtered_no_matches
+6-9,Jobs/jobs_page_type_filtered
+6-9,Jobs/jobs_page_type_filtered_no_matches
 3,Jobs/show_job
-6-8,Jobs/show_jobs
+6-9,Jobs/show_jobs
 3,ORMQueries/activerecord_type_introspection_query
 0,ORMQueries/asyncpg_types
 0,ORMQueries/column_descriptions_json_agg

